src/store/actions/productsActions/getProducts.ts

```diff
@@ -0,0 +1,36 @@
+import * as actionTypes from "../../actionTypes/productsActionTypes";
+import * as productsInterfaces from "../../types/productsInterfaces";
+
+import { getProductsRequest } from "../../../api/productsApi";
+import { Dispatch } from "react";
+
+// get products
+const setProductsSucces = ({
```

Comment:
I think here better will be if you create only one const it will be object, and in this object create you methods: setProductsSucces, setProductsFail, getProducts
and add method for pending request. its need for spinner.